packages/dbml-cli/src/connectors/MssqlConnector.js

```diff
@@ -0,0 +1,508 @@
+            END AS column_default,
+            -- Fetching table comments
+            p.value AS table_comment,
+            ep.value AS column_comment
+        FROM
+            sys.tables t
+        JOIN
+            sys.schemas s ON t.schema_id = s.schema_id
+        JOIN
+            sys.columns c ON t.object_id = c.object_id
+        JOIN
+            sys.types ty ON c.user_type_id = ty.user_type_id
+        LEFT JOIN
+            sys.extended_properties p ON p.major_id = t.object_id
+                AND p.name = 'MS_Description'
+                AND p.minor_id = 0 -- Ensure minor_id is 0 for table comments
+        LEFT JOIN
+            sys.extended_properties ep ON ep.major_id = c.object_id
+                AND ep.minor_id = c.column_id
+                AND ep.name = 'MS_Description'
+        WHERE
+            t.type = 'U' -- User-defined tables
+    )
+    SELECT
+        tf.table_schema,
+        tf.table_name,
+        tf.column_name,
+        tf.data_type,
+        tf.character_maximum_length,
+        tf.numeric_precision,
+        tf.numeric_scale,
+        tf.identity_increment,
+        tf.is_nullable,
+        tf.column_default,
+        tf.table_comment,
+        tf.column_comment,
+        cc.name AS check_constraint_name, -- Adding CHECK constraint name
+        cc.definition AS check_constraint_definition, -- Adding CHECK constraint definition
+        CASE
+            WHEN tf.column_default LIKE '((%))' THEN 'number'
+            WHEN tf.column_default LIKE '(''%'')' THEN 'string'
+            ELSE 'expression'
+        END AS default_type
+    FROM
+        tables_and_fields AS tf
+    LEFT JOIN
+        sys.check_constraints cc ON cc.parent_object_id = OBJECT_ID(tf.table_schema + '.' + tf.table_name)
+        AND cc.definition LIKE '%' + tf.column_name + '%' -- Ensure the constraint references the column
+    ORDER BY
+        tf.table_schema,
+        tf.table_name,
+        tf.column_name;
+  `;
+
+  const tablesAndFieldsResult = await client.query(tablesAndFieldsSql);
+  const tables = tablesAndFieldsResult.recordset.reduce((acc, row) => {
+    const {
+      table_schema,
+      table_name,
+      table_comment,
+      check_constraint_name,
+      check_constraint_definition,
+    } = row;
+
+    if (!acc[table_name]) {
+      acc[table_name] = {
+        name: table_name,
+        schemaName: table_schema,
+        note: table_comment ? { value: table_comment } : { value: '' },
+      };
+    }
+
+    const enumValues = getEnumValues(check_constraint_definition);
+    if (enumValues) {
+      enums.push({
+        name: check_constraint_name,
+        schemaName: table_schema,
+        values: enumValues,
+      });
+    }
+
+    if (!fields[table_name]) fields[table_name] = [];
+    const field = generateField(row);
+    if (enumValues) {
+      field.type = {
+        type_name: check_constraint_name,
+        schemaName: table_schema,
+      };
+    }
+    fields[table_name].push(field);
```

Comment:
Will this work the case where we have the many table with the same name but in multiple schema? From reading the code, I believe we can't distinguish which `schema.table` the field belong to later on when we passed the `fields array` to `DatabaseGenerator`. For this to work, I think you could use a key like: `schemaName.tableName`

Also please check for this case with your postgres connector as well.

Reply:
Fix this issue for both postgresql and mssql nha.